src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternParser.cs

```diff
@@ -0,0 +1,573 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Routing.Patterns
+{
+    internal static class RoutePatternParser
```

Comment:
This is also another rename of an existing class, the changes here are to support the `RoutePattern` instead of `RouteTemplate`.